js/app/src/css.ts

```diff
@@ -18,3 +18,39 @@ export function mount_css(url: string, target: HTMLElement): Promise<void> {
 		target.appendChild(link);
 	});
 }
+
+export function prefix_css(
+	string: string,
+	version: string,
+	style_element = document.createElement("style")
+): HTMLStyleElement {
+	//  create style element
+	style_element.remove();
+
+	const stylesheet = new CSSStyleSheet();
+	stylesheet.replaceSync(string);
+
+	const rules = stylesheet.cssRules;
+
+	let css_string = "";
+
+	for (let i = 0; i < rules.length; i++) {
+		const rule = rules[i];
+
+		if (rule instanceof CSSStyleRule) {
+			const selector = rule.selectorText;
+			if (selector) {
+				const new_selector = selector
+					.split(",")
+					.map((s) => `.gradio-container-${version} .contain ${s.trim()}`)
+					.join(",");
+
+				css_string += rule.cssText.replace(selector, new_selector);
+			}
```

Comment:
Lets leave the existing styles in as well, as they are, in case they are targetting something outside of the gradio app + to avoid breaking changes.